p2p/discover/v5_udp.go

```diff
@@ -359,21 +400,31 @@ func lookupDistances(target, dest enode.ID) (dists []uint) {
 
 // ping calls PING on a node and waits for a PONG response.
 func (t *UDPv5) ping(n *enode.Node) (uint64, error) {
+	pong, err := t.PingWithResp(n)
+	if err != nil {
+		return 0, err
+	}
+
+	return pong.ENRSeq, nil
+}
+
+// PingWithResp calls PING on a node and waits for a PONG response.
+func (t *UDPv5) PingWithResp(n *enode.Node) (*v5wire.Pong, error) {
```

Comment:
We can just change `Ping` signature to return the response object. No need to add `WithoutResp` method.